File: Mobile-app/lib/utils/supabase_embeds.dart
/// PostgREST embed hints — required after [event_faces.best_match_resident_id]
/// (multiple FKs between event_faces and residents).
class SupabaseEmbeds {
  SupabaseEmbeds._();

  static const String eventFaces =
      'id, classification, match_score, resident_id, unknown_profile_id, '
      'camera_event_id, best_match_resident_id, '
      'residents!resident_id(name, id), '
      'unknown_face_profiles(id, display_label, sighting_count, first_seen_at, status)';

  static const String cameraEvent =
      '*, events(*), event_faces($eventFaces)';

  static const String eventFaceRow =
      'id, classification, match_score, resident_id, unknown_profile_id, '
      'camera_event_id, best_match_resident_id, '
      'camera_events(id, snapshot_path, created_at, event_id), '
      'residents!resident_id(name), '
      'unknown_face_profiles(id, display_label, sighting_count, first_seen_at, status)';
}

File: website/client/src/utils/supabaseEmbeds.js
/**
 * PostgREST embed hints — required after event_faces.best_match_resident_id FK
 * (multiple relationships between event_faces and residents).
 */
export const EVENT_FACES_EMBED =
  'id, classification, match_score, resident_id, unknown_profile_id, camera_event_id, best_match_resident_id, ' +
  'residents!resident_id(name), ' +
  'best_match:residents!best_match_resident_id(id, name), ' +
  'unknown_face_profiles(id, display_label, sighting_count, first_seen_at, status)';

export const CAMERA_EVENT_EMBED =
  `*, events(*), event_faces(${EVENT_FACES_EMBED})`;
